src/layers/domain/core/error.py

```diff
@@ -14,6 +14,10 @@ class BadProductIdError(Exception):
     pass
 
 
+class BadOdsCodeError(Exception):
```

Comment:
a lot of the errors and associated validators in `validation.py` are now orphaned due to this PR:

* BadProductIdError
* InvalidOdsCodeError
* BadAsidError
* BadOdsCodeError

Reply:
While I can easily find unused libraries and private scope references, how do you locate public and even entire files that are unused?

Reply:
i'm not sure it's even possible in python, unfortunately relies on manual book keeping